### ISSUE TITLE
Question about differences from lmer/lme

### ISSUE THREAD

**Issue by itissid** [paul-buerkner/brms#214]
TL;DR: Why is there a difference in the way the contrasts work for brm vs lme/lmer?
The answer may be trivial/inconsequential, sorry if it seems noobish as this is all new to me.

I was playing with an example with a data set for schools. Here is the creation of the data set and its fit in lmer,lme  and brms:

```
library(brms)
library(nlme)
library(lme4)

# Pull school level data.
mses <- with(MathAchieve, tapply(SES, School, mean))
sector <- MathAchSchool$Sector
#Pull the names of the sectors from school level data
names(sector) <- row.names(MathAchSchool)
Bryk <- as.data.frame(MathAchieve[, c("School", "SES", "MathAch")])
names(Bryk) <- tolower(names(Bryk))
#Add mean ses and sector to the main data frame
Bryk <- within(Bryk, {
 meanses <- as.vector(mses[as.character(school)])
 cses <- ses - meanses
 sector <- sector[as.character(school)]
})
Bryk$sector <- factor(Bryk$sector, levels=c("Public", "Catholic"))
set.seed(12345)
bryk.lmer.1 <- lmer(mathach ~ meanses*cses + sector*cses+ (cses | school),
 data=Bryk)
bryk.lme.1 <- lme(mathach ~ meanses*cses + sector*cses, random= ~ cses | school,
 data=Bryk)

formulae = mathach ~ meanses*cses + sector*cses+ (1 + cses | school)
bryk.brm.1 <- brm(formulae,
 data=Bryk, 
 cores=parallel::detectCores()/2, control=list(adapt_delta=0.99), iter=4000,
 prior = c(set_prior("normal(0,5)", class = "b"),
 set_prior("cauchy(0,2)", class = "sd"),
 set_prior("lkj(2)", class = "cor")))
```

The lmer and lme model work ok, but with brm I got the error:

```
Error in ctrfn(levels(x), contrasts = contrasts) :
  orthogonal polynomials cannot be represented accurately enough for 159 degrees of freedom
```
Which is an error that comes from the contrasts function:
```
> contrasts(Bryk$school)
Error in ctrfn(levels(x), contrasts = contrasts) :
  orthogonal polynomials cannot be represented accurately enough for 159 degrees of freedom
```

The error was occurring for the school factor that was ordered. If I make it unordered it works:
```
Bryk$school = factor(Bryk$school, ordered=F)
```

So again why is there a difference in the way the contrasts work for brm vs lme/lmer?

**Comment by paul-buerkner** [comment]
It is not really a difference about how contrasts work, but that apparently that brms can't handle ordered factors as grouping terms. I think this should not be too difficult to fix.

I have a question though: Why should one use an ordered factor as a grouping term?

**Comment by paul-buerkner** [comment]
On the dev version of brms on github, it should now be working correctly! :-)

**Comment by itissid** [comment]
@paul-buerkner Thanks for the fix. And to answer your Q, yes using ordered factors as grouping terms is kinda odd. But I suspect because brms should not really care about that you made that change.

**Comment by paul-buerkner** [comment]
Indeed. Maybe it is worth giving a little bit of explanation on why this problem happend in the first place:
People may choose to use the global contrasts option to temporalily change the default contrast that is used when no explicit contrast is specified. However, when they use their fitted model in a fresh R session, the contrast option may be different leading to problems in many post-processing options. Thus, brms saves the contrasts used at the time of model fitting. Previously I did this for all factors, but for factors used only as grouping variables, there is no real need to to it, in addition to causing problems with some contrasts options as we have seen.
